spdx-utils/src/test/kotlin/SpdxExpressionTest.kt

```diff
@@ -367,6 +369,84 @@ class SpdxExpressionTest : WordSpec() {
             }
         }
 
+        "applyChoice()" should {
+            "return the choice for a simple SPDX expression" {
+                val spdxExpression = "a".toSpdx()
+
+                val resolvedLicense = spdxExpression.applyChoice(spdxExpression, "a".toSpdx())
+
+                resolvedLicense shouldBe "a".toSpdx()
+            }
+
+            "return the expression if the matcher does not match the expression" {
+                val spdxExpression = "a".toSpdx()
+
+                val resolvedLicense = spdxExpression.applyChoice("b".toSpdx(), "c".toSpdx())
+
+                resolvedLicense shouldBe "a".toSpdx()
+            }
+
+            "throw an exception if the user chose a wrong license for a simple SPDX expression" {
+                val spdxExpression = "a".toSpdx()
+
+                assertThrows(InvalidLicenseChoiceException::class.java) {
+                    spdxExpression.applyChoice(spdxExpression, "b".toSpdx())
+                }
+            }
+
+            "return the choice if it is a valid choice" {
+                val spdxExpression = "a OR b OR c".toSpdx()
+
+                val resolvedLicense = spdxExpression.applyChoice(spdxExpression, "b".toSpdx())
+
+                resolvedLicense shouldBe "b".toSpdx()
+            }
+
+            "throw an exception if the chosen license is not an valid option" {
+                val spdxExpression = "a OR b OR c".toSpdx()
+
+                assertThrows(InvalidLicenseChoiceException::class.java) {
+                    spdxExpression.applyChoice(spdxExpression, "d".toSpdx())
+                }
+            }
+            
+            "return the expression if the matcher does not match the whole expression" {
+                val spdxExpression = "a OR b OR c".toSpdx()
+                val expressionMatcher = "b OR c".toSpdx()
+
+                val resolvedLicense = spdxExpression.applyChoice(expressionMatcher, "b".toSpdx())
+
+                resolvedLicense shouldBe "a OR b OR c".toSpdx()
```

Comment:
Not really sure about that one.
Is `b OR c` a valid subtree of `a OR b OR c` and therefore `b` would be a valid choice?

Reply:
Yes, I would say so.